src/scipp/core/bins.py

```diff
@@ -124,6 +124,19 @@ def lookup(
     return Lookup(_cpp.lookup_previous, func, dim, fill_value)
 
 
+class Constituents(TypedDict):
+    """A dict with bin constituents."""
+
+    data: Variable | DataArray
+    """Buffer with events."""
```

Comment:
```suggestion
    """Data content"""
```
avoid "events", since that is Neutron terminology.